### PR TITLE
openpmd-api: add v0.16.0

### PR DESCRIPTION
**Pull request by ax3l**
Add the latest release of openPMD-api.

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@13 openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@12 openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@11 openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@10 openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@9 openpmd-api@0.16.0~mpi~hdf5~adios2%gcc@8
-- linux-ubuntu22.04-skylake / gcc@8.4.0 ------------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@9.5.0 ------------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
openpmd-api@0.16.0~adios2~hdf5~ipo~mpi~python+shared build_system=cmake build_type=Release generator=make
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
./build-quality-tools/build_pr_changes.py -c gcc@13,gcc@12,gcc@11,gcc@10,gcc@9,gcc@8 -b openpmd-api@0.16.0~mpi~hdf5~adios2 -ay
```

**Review by franzpoeschel — APPROVED** [review]

**Comment by ax3l** [comment]
@bernhardkaindl ready for merge :)

File: var/spack/repos/builtin/packages/openpmd-api/package.py [hunk 1]
```diff
@@ -45,18 +46,19 @@ class OpenpmdApi(CMakePackage):
...
     depends_on("cmake@3.15.0:", type="build")
+    depends_on("cmake@3.22.0:", type="build", when="@0.16.0:")
     depends_on("catch2@2.6.1:2", type="test")
     depends_on("catch2@2.13.4:2", type="test", when="@0.14.0:")
     depends_on("catch2@2.13.10:2", type="test", when="@0.15.0:")
     depends_on("mpi@2.3:", when="+mpi")  # might become MPI 3.0+
     depends_on("nlohmann-json@3.9.1:")
     depends_on("mpark-variant@1.4.0:", when="@:0.14")  # pre C++17 releases
-    depends_on("toml11@3.7.1:3.8.1", when="@0.15.0:")
+    depends_on("toml11@3.7.1:3", when="@0.15.0:")
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/openpmd-api/package.py]: Ah, we support 4.0+ now, too:
```suggestion
    depends_on("toml11@3.7.1:3", when="@0.15.0:")
    depends_on("toml11@3.7.1:", when="@0.16.0:")
```

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants openpmd-api
-- linux-ubuntu22.04-x86_64_v3 / gcc@11.4.0 ---------------------
openpmd-api@0.16.0~adios2+hdf5~ipo+mpi~python+shared build_system=cmake build_type=Release generator=make patches=e49fe79
openpmd-api@0.16.0+adios2+hdf5~ipo+mpi~python+shared build_system=cmake build_type=Release generator=make patches=e49fe79
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
./build-quality-tools/./build_pr_changes.py -b openpmd-api
```

**Pull request merged**